### PR TITLE
agents/check_mk_agent.openwrt: do not use pushd/popd

### PR DESCRIPTION
**Pull request by sch-m**
These are built-in commands of bash and not POSIX shell conform.

Signed-off-by: Martin Schiller <ms@dev.tdt.de>

### CODE REVIEW

**Comment by jherbel** [comment]
Hello Martin,

thank you for this pull request. We will have a closer look at your patch and discuss internally on how and if we can implement it into the official code base.

Internal Ref: CMK-4970

File: agents/check_mk_agent.openwrt [hunk 1]
```diff
@@ -336,9 +336,10 @@ fi
 # Current state of bonding interfaces
 if [ -e /proc/net/bonding ] ; then
     echo '<<<lnx_bonding:sep(58)>>>'
-    pushd /proc/net/bonding > /dev/null
+    olddir=$(pwd)
```
Comment (rawiriblundell) [inline agents/check_mk_agent.openwrt:339]: Hi,
Have you considered/tried a subshell instead? e.g.

    # Current state of bonding interfaces
    if [ -e /proc/net/bonding ]; then
        echo '<<<lnx_bonding:sep(58)>>>'
        (
            cd /proc/net/bonding || return
            head -v -n 1000 ./*
        )
    fi

Note that this example code (that I wrote over a year ago) also corrects two other issues that Shellcheck raises

* https://github.com/koalaman/shellcheck/wiki/SC2164
* https://github.com/koalaman/shellcheck/wiki/SC2035

Comment (sch-m) [reply agents/check_mk_agent.openwrt:339]: You are right, this would also work.

**Comment by sch-m** [comment]
I've updated the commit to follow Rawiri Blundell's suggestions, but in a slightly other way to keep the existing indentation.

**Comment by sch-m** [comment]
I don't know why the CI check now fails at the "test-unit" stage, but I don't think that this is related to my last changes when I have a look at the output of the check.

When I try to run the test locally, i get this outut:
`Warning: Python 3.8 was not found on your system…
You can specify specific versions of Python with:
  $ pipenv --python path/to/python
`

**Comment by rawiriblundell** [comment]
Yeah this happens all the time here.  Python failures for sh code submissions... makes perfect sense :)

**Comment by jherbel** [comment]
Hi,
the issue with the unit tests was a bug from our side, if you want, you can rebase onto the current master, this should resolve it, however, the PR is queued for review anyway, so you don't have to.
Best
Jörg

**Pull request closed without merging**